libvast/src/system/evaluator.cpp

```diff
@@ -153,7 +153,17 @@ evaluator(evaluator_actor::stateful_pointer<evaluator_state> self,
   self->state.expr = std::move(expr);
   self->state.eval = std::move(eval);
   self->state.store = std::move(store);
+  self->send(self->state.store, atom::exporter_v,
+             caf::actor_cast<caf::actor>(self));
   return {
+    [self](table_slice slice) {
+      self->send(self->state.client, std::move(slice));
+    },
+    [self](atom::done, caf::error) {
+      VAST_DEBUG("{} completed expression evaluation", self);
+      self->state.promise.deliver(atom::done_v);
```

Comment:
How we we handle errors here?

Reply:
I don't think escalating makes sense. I'd say just log `VAST_ERROR` and move on.

Reply:
Okay, was mostly wondering because we could deliver the error to the promise instead of the done.